text/2018-10-25-batch-split.md

```diff
@@ -0,0 +1,170 @@
+# Summary
+
+Support `BatchSplit` feature that split one Region into multiple Regions at a time if the size is large enough. This includes modifications of both TiKV and PD. For TiKV, every round of split-check produces multiple split keys instead of one and change inner split related interface into batch style. For PD, add RPC `AskBatchSplit` and `ReportBatchSplit`  to permit TiKV asking for `region_id` and `peer_id` in batch.
```

Comment:
not only size is large enough.

We may split a region if the number of keys exceeds a threshold too.